examples/sphere/opt_limiters_solidbody.jl

```diff
@@ -0,0 +1,331 @@
+            if rd[1] <= r0 && abs(λ - centers[1].long) * R >= rad2deg(r0 / 6)
+                q = 1.0
+            elseif rd[2] <= r0 &&
+                   abs(λ - centers[2].long) * R >= rad2deg(r0 / 6)
+                q = 1.0
+            elseif rd[1] <= r0 &&
+                   abs(λ - centers[1].long) * R < rad2deg(r0 / 6) &&
+                   (ϕ - centers[1].lat) * R < rad2deg(-5 * r0 / 12)
+                q = 1.0
+            elseif rd[2] <= r0 &&
+                   abs(λ - centers[2].long) * R < rad2deg(r0 / 6) &&
+                   (ϕ - centers[2].lat) * R > rad2deg(5 * r0 / 12)
+                q = 1.0
+            else
+                q = 0.1
+            end
+        elseif test_name == gaussian_test_name
+            q = 0.95 * (exp(-5.0 * (rd[1] / r0)^2) + exp(-5.0 * (rd[2] / r0)^2))
+        else # default test case, cosine bells
+            if rd[1] < r0
+                q = 0.1 + 0.9 * (1 / 2) * (1 + cospi(rd[1] / r0))
+            elseif rd[2] < r0
+                q = 0.1 + 0.9 * (1 / 2) * (1 + cospi(rd[2] / r0))
+            else
+                q = 0.1
+            end
+        end
+
+        # Initialize air density
+        ρ = 1.0
+
+        # Tracer density
+        Q = ρ * q
+        return (ρ = ρ, ρq = Q)
+    end
+
+    function f!(ystar, y, parameters, t)
+
+        grad = Operators.Gradient()
+        wdiv = Operators.WeakDivergence()
+        T = parameters.T
+
+        coords = Fields.coordinate_field(axes(y.ρq))
+        u = map(coords) do coord
+            ϕ = coord.lat
+            λ = coord.long
+
+            uu =
+                u0 * sind(λ)^2 * sind(2 * ϕ) * cospi(t / T) +
+                360.0 * cosd(ϕ) / T
+            uv = u0 * sind(2 * λ) * cosd(ϕ) * cospi(t / T)
+            Geometry.UVVector(uu, uv)
+        end
+
+        # Compute min_Q[] and max_Q[] that will be needed later in the stage limiter
+        space = parameters.space
+        n_elems = Topologies.nlocalelems(space)
+        topology = space.topology
+
+        neigh_elems_Q_min = Array{Float64}(undef, 8)
+        neigh_elems_Q_max = Array{Float64}(undef, 8)
+
+        for e in 1:n_elems
+            Q_e = Fields.slab(y.ρq, e)
+
+            Q_e_min = minimum(Q_e)
+            Q_e_max = maximum(Q_e)
+            neigh_elems = Topologies.neighboring_elements(topology, e)
+            for i in 1:length(neigh_elems)
+                if neigh_elems[i] == 0
+                    neigh_elems_Q_min[i] = +Inf
+                    neigh_elems_Q_max[i] = -Inf
+                else
+                    neigh_elems_Q_min[i] =
+                        Fields.minimum(Fields.slab(y.ρq, neigh_elems[i]))
+                    neigh_elems_Q_max[i] =
+                        Fields.maximum(Fields.slab(y.ρq, neigh_elems[i]))
+                end
+            end
+            parameters.min_Q[e] = min(minimum(neigh_elems_Q_min), Q_e_min)
+            parameters.max_Q[e] = max(maximum(neigh_elems_Q_max), Q_e_max)
+        end
```

Comment:
Probably not in this PR, but we should add a function that computes these (`neighbour_extrema`?)

Reply:
Yeah. I would leave it to another PR